src/features/henHouse/HenHouseInside.tsx

```diff
@@ -58,13 +67,21 @@ export const HenHouseInside: React.FC = () => {
 
   mapPlacements.push(...components);
 
+  const nextLevel = Math.min(level + 1, 3) as Exclude<AnimalBuildingLevel, 1>;
+
   return (
     <>
       <AnimalBuildingModal
         buildingName="Hen House"
         show={showModal}
         onClose={() => setShowModal(false)}
       />
+      <UpgradeBuildingModal
+        buildingName="Hen House"
+        level={nextLevel}
```

Comment:
I wasn't able to upgrade to level 3 with this configuration. It showed up that I was at max level when I was only level 2. I manually bypassed it to test the final level.

Reply:
It was because you needed to bomb your DB. Old data was initialised with a building at Level 0.

Can you confirm you deleted es henhouse and barn in your DB before testing?